app/transactions/members/update_transaction.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class Members::UpdateTransaction
+  include Dry::Transaction
+
+  step :extract_ids
+  step :assign_static_slot
+  step :update_member
+
+  private
+
+  def extract_ids(input)
+    return Success(input) if input[:permitted_params][:static_slots_attributes].nil?
+
+    static_slot_ids = []
+    input[:permitted_params][:static_slots_attributes].values.each do |value|
+      static_slot_ids << value['id']
+    end
+    Success(input.merge({ static_slot_ids: static_slot_ids }))
+  end
+
+  def assign_static_slot(input)
+    return Success(input) if input[:permitted_params][:static_slots_attributes].nil?
+
+    input[:static_slot_ids].each do |static_slot_id|
+      unless ::StaticSlotMember.create(member_id: input[:current_member].id, static_slot_id: static_slot_id)
+        Failure(error: t('activerecord.errors.models.mission.messages.static_slot_extraction_failure'))
```

Comment:
I guess that you meant `static_slot_creation_failure` instead of `static_slot_extraction_failure` ?

By the way, is it possible to use the I18n shorthand `t('.static_slot_creation_failure')` here?

Reply:
?